transportable-udfs-trino/src/main/java/com/linkedin/transport/trino/StdUdfWrapper.java

```diff
@@ -141,17 +142,17 @@ public ScalarFunctionImplementation specialize(FunctionBinding functionBinding,
         - (new Random()).nextInt(initialJitterInt));
     boolean[] nullableArguments = stdUDF.getAndCheckNullableArguments();
 
-    return new ChoicesScalarFunctionImplementation(
-        functionBinding,
+    return new ChoicesSpecializedSqlScalarFunction(
+        boundSignature,
         NULLABLE_RETURN,
         getNullConventionForArguments(nullableArguments),
-        getMethodHandle(stdUDF, functionBinding, nullableArguments, requiredFilesNextRefreshTime));
+        getMethodHandle(stdUDF, boundSignature, nullableArguments, requiredFilesNextRefreshTime));
   }
 
-  private MethodHandle getMethodHandle(StdUDF stdUDF, FunctionBinding functionBinding, boolean[] nullableArguments,
+  private MethodHandle getMethodHandle(StdUDF stdUDF, BoundSignature boundSignature, boolean[] nullableArguments,
```

Comment:
Seems like `functionBinding` is available in the caller method. 
Since `getMethodHandle(...)` is a private utility, is the change from `functionBinding` -> `boundSignature` necessary or good to have?